### ISSUE TITLE
"pipx install --force --python ..." doesn't change Python version

### ISSUE THREAD

**Issue by kshpytsya** [pypa/pipx#447]
```
$ pipx install --python python3.8 tinyshar
  installed package tinyshar 0.10.2, Python 3.8.3
  These apps are now globally available
    - tinyshar
done! ✨ 🌟 ✨
$ pipx install --force --python python3.7 tinyshar
Installing to existing directory '/home/uken/.local/pipx/venvs/tinyshar'
  installed package tinyshar 0.10.2, Python 3.8.3
  These apps are now globally available
    - tinyshar
done! ✨ 🌟 ✨
$ pipx uninstall tinyshar
uninstalled tinyshar! ✨ 🌟 ✨
$ pipx install --python python3.7 tinyshar
  installed package tinyshar 0.10.2, Python 3.7.8
  These apps are now globally available
    - tinyshar
done! ✨ 🌟 ✨
```

Observe the second invocation completing successfully yet essentially silently ignoring the `--python` flag.

In case of Python version specified via `--python` not matching version in the existing venv, I would expect `pipx install --force --python ...` to either recreate the venv, or fail with an error suggesting user to do `pip uninstall <pkg> && pip install --python <python> <pkg>`.

**Comment by cs01** [comment]
The `--verbose` flag will show all commands being run. In this case, there is already an existing venv in which pip is being run. Here is some example output:
```
> pipx install youtube-dl --force --verbose
pipx > (package_name_from_spec:216): Determined package name: youtube-dl
pipx > (package_name_from_spec:217): Package name determined in 0.0s
Installing to existing directory '/home/csmith/.local/pipx/venvs/youtube-dl'
pipx > (needs_upgrade:52): Time since last upgrade of shared libs, in seconds: 9567.118878602982. Upgrade will be run by pipx if greater than 2592000.0.
pipx > (run_subprocess:112): running /usr/bin/python3.7 -m venv --without-pip /home/csmith/.local/pipx/venvs/youtube-dl
pipx > (run_subprocess:112): running /home/csmith/.local/pipx/venvs/youtube-dl/bin/python -c import sysconfig; print(sysconfig.get_path('purelib'))
pipx > (run_subprocess:112): running /home/csmith/.local/pipx/shared/bin/python -c import sysconfig; print(sysconfig.get_path('purelib'))
pipx > (run_subprocess:112): running /home/csmith/.local/pipx/venvs/youtube-dl/bin/python --version
pipx > (run_subprocess:112): running /home/csmith/.local/pipx/venvs/youtube-dl/bin/python -m pip install youtube-dl
Requirement already satisfied: youtube-dl in /home/csmith/.local/pipx/venvs/youtube-dl/lib/python3.7/site-packages (2020.6.16.1)
pipx > (run_subprocess:112): running /home/csmith/.local/pipx/venvs/youtube-dl/bin/python -c <contents of venv_metadata_inspector.py> youtube-dl /home/csmith/.local/pipx/venvs/youtube-dl/bin
pipx > (_symlink_package_apps:78): Force is true. Removing /home/csmith/.local/bin/youtube-dl.
pipx > (needs_upgrade:52): Time since last upgrade of shared libs, in seconds: 9567.716672897339. Upgrade will be run by pipx if greater than 2592000.0.
  installed package youtube-dl 2020.6.16.1, Python 3.7.7
  These apps are now globally available
    - youtube-dl
done! ✨ 🌟 ✨
```
In this case, the venv already exists and so does the python interpreter, so pip just re-uses it. The force flag is to indicate the existing venv is to be used and any symlinks created by pipx in the bin dir will be overwritten.

It sounds like a `pipx reinstall tinyshar` would be more appropriate, but `reinstall` doesn't exist (it probably should be added).

```
pipx uninstall tinyshar
pipx install tinyshar --python python3.8
```
will do it though.

**Comment by hotenov** [comment]
>  but reinstall doesn't exist (it probably should be added).

I'm from the future :) `reinstall` command was added in `v0.16.0.0` _(Jan 16, 2021)_

`pipx upgarde --force package` also does not upgrade existing venv in the end of 2022

but now (as mention above) you can use one command `pipx reinstall package` for upgrading as venv as package itself

**Comment by georgek** [comment]
I'm from even further in the future, and this behaviour is still a bit confusing, but the original issue seems fixed.

I wanted to reinstall a package into the latest version of Python, so I issued `pipx reinstall <package>`. The package was reinstalled into the same Python version.

I tried to then do `pipx reinstall <package> --python python3.11` but it just said "Nothing to reinstall for <package> 😴"

There is no `--force` flag for `reinstall`.

I was almost going to uninstall and install, then saw this and tried `pipx install --force <package> --python python3.10` and it worked!

So now I'm confused what the `reinstall` command is for...

**Comment by dukecat0** [comment]
This issue still exists, and `pipx reinstall <package> --python python3.10` works for me:
```
➜  ~ pipx install --force mat2 --python python3.10
Installing to existing venv 'mat2'
  installed package mat2 0.13.4, installed using Python 3.9.2
  These apps are now globally available
    - mat2
done! ✨ 🌟 ✨
➜  ~ pipx reinstall mat2 --python python3.10                
uninstalled mat2! ✨ 🌟 ✨
  installed package mat2 0.13.4, installed using Python 3.10.1
  These apps are now globally available
    - mat2
done! ✨ 🌟 ✨
```

**Comment by georgek** [comment]
My apologies, I had an environment mix up and the above behaviour I posted was for pipx 0.16.4. With the current version 1.2.0 I see the same behaviour as @dukecat0.

My version was newer than this issue, though, so it seems like it was fixed but has regressed again since 0.16.4.

**Comment by uranusjr** [comment]
To be sure, it was fixed in 0.16.4 (or somewhere around it), broken again, and fixed again some version before the current one. So the feature is working now, right?
